src/zabapgit_object_clas.prog.abap

```diff
@@ -875,202 +1170,671 @@ CLASS lcl_object_clas IMPLEMENTATION.
+      mt_sotr_package          TYPE devclass,
+      mv_docu_object_name      TYPE dokhl-object,
+      mv_docu_language         TYPE spras,
+      mt_docu_lines            TYPE tlinetab.
+
+ENDCLASS.
+CLASS ltd_spy_oo_object IMPLEMENTATION.
+  METHOD lif_object_oriented_object~create.
+    DATA lv_properties_structure_name TYPE string.
+    lv_properties_structure_name = cl_abap_typedescr=>describe_by_data( is_properties )->absolute_name.
+    IF lv_properties_structure_name = cl_abap_typedescr=>describe_by_data( ms_interface_properties )->absolute_name.
+      ms_interface_properties = is_properties.
+    ELSE.
+      ms_class_properties     = is_properties.
+    ENDIF.
+    mv_package                = iv_package.
+    mv_overwrite              = iv_overwrite.
+  ENDMETHOD.
+  METHOD lif_object_oriented_object~generate_locals.
+    ms_locals_key            = is_key.
+    mt_local_definitions     = it_local_definitions.
+    mt_local_implementations = it_local_implementations.
+    mt_local_macros          = it_local_macros.
+    mt_local_test_classes    = it_local_test_classes.
+    mv_force                 = iv_force.
+  ENDMETHOD.
 
-      WHEN 'INTF'.
-        io_xml->read( EXPORTING iv_name = 'VSEOINTERF'
-                      CHANGING cg_data = ls_vseointerf ).
+  METHOD lif_object_oriented_object~deserialize_source.
+    ms_deserialize_key = is_key.
+    mt_source          = it_source.
+  ENDMETHOD.
 
-        CALL FUNCTION 'SEO_INTERFACE_CREATE_COMPLETE'
-          EXPORTING
-            devclass        = iv_package
-            overwrite       = seox_true
-          CHANGING
-            interface       = ls_vseointerf
-          EXCEPTIONS
-            existing        = 1
-            is_class        = 2
-            db_error        = 3
-            component_error = 4
-            no_access       = 5
-            other           = 6
-            OTHERS          = 7.
-        IF sy-subrc <> 0.
-          lcx_exception=>raise( 'Error from SEO_INTERFACE_CREATE_COMPLETE' ).
-        ENDIF.
+  METHOD lif_object_oriented_object~add_to_activation_list.
+    ms_item_to_activate = is_item.
+  ENDMETHOD.
+
+  METHOD lif_object_oriented_object~update_descriptions.
+    ms_description_key = is_key.
+    mt_descriptions    = it_descriptions.
+  ENDMETHOD.
+
+  METHOD lif_object_oriented_object~insert_text_pool.
+    mv_text_pool_inserted = abap_true.
+    mv_text_pool_class_name = iv_class_name.
+    mt_text_pool = it_text_pool.
+    cl_abap_unit_assert=>assert_equals(
+      act = iv_language
+      exp = sy-langu ).
+  ENDMETHOD.
+
+  METHOD lif_object_oriented_object~create_sotr.
+    mt_sotr = it_sotr.
+    mt_sotr_package = iv_package.
+  ENDMETHOD.
+
+  METHOD lif_object_oriented_object~create_documentation.
+    mv_docu_object_name = iv_object_name.
+    mv_docu_language    = iv_language.
+    mt_docu_lines       = it_lines.
+  ENDMETHOD.
 
+ENDCLASS.
+
+CLASS ltd_fake_object_files DEFINITION FOR TESTING
```

Comment:
Yes, that can be done. Should I move everything to the unit test include or rather create a new test include?